packages/smoldot-discovery/package.json

```diff
@@ -0,0 +1,30 @@
+{
+  "name": "@substrate/smoldot-discovery",
+  "version": "0.0.6",
+  "author": "Parity Technologies (https://github.com/paritytech)",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/paritytech/substrate-connect.git"
+  },
+  "license": "MIT",
+  "exports": {
+    "./api": "./dist/api/index.mjs",
+    "./smoldot": "./dist/smoldot/index.mjs"
```

Comment:
To check my understanding, will this return something that you call like so:

```ts
import { .. } from "@substrate/smoldot-discoery/api"
import { .. } from "@substrate/smoldot-discoery/smoldot"
```

I think that as it stands, you'd need to include both this package and "@substrate/discovery" to use this, and then do something like:

```ts
import { getProviders } from "@substrate/discovery"
import { isSmoldotExtension, SmoldotExtensionProviderDetail } from "@substrate/smoldot-discovery/api"

// Not actually certain if this ends up with the provider typing or not:
let smoldotProviders: SmoldotExtensionProviderDetail[] = getProviders().filter(isSmoldotExtension)
```

Personally I think, to make this as simple as possible to use, we'd want to hand something back like:

```ts
// Most dApps will just import this (and maybe the types):
import { getSmoldotProviders, SmoldotProvider } from "@substrate/smoldot-discovery"

let smoldotProviders: SmoldotProvider[] = getSmoldotProviders()
```

And then I geuss extensions would just want to import the types to use so might do something like:

```ts
// Extensions might import just the types to make sure they expose a compliant interface:
import { SmoldotProvider, WellKnownChain .. } from "@substrate/smoldot-discovery/types"
```